contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1767,43 +1770,61 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
     /**
      * Initializes the execution context.
      * @param _transaction OVM transaction being executed.
+     * Security considerations: 
+     * It is critical that these values are initialized to their safe defaults in the `_initContext` function. Thus all 
+     * types other than enums need to be set to zero.
      */
     function _initContext(
         Lib_OVMCodec.Transaction memory _transaction
     )
         internal
     {
+        transactionContext.ovmL1QUEUEORIGIN = _transaction.l1QueueOrigin;
         transactionContext.ovmTIMESTAMP = _transaction.timestamp;
         transactionContext.ovmNUMBER = _transaction.blockNumber;
+        transactionContext.ovmGASLIMIT = gasMeterConfig.maxGasPerQueuePerEpoch;
         transactionContext.ovmTXGASLIMIT = _transaction.gasLimit;
-        transactionContext.ovmL1QUEUEORIGIN = _transaction.l1QueueOrigin;
         transactionContext.ovmL1TXORIGIN = _transaction.l1TxOrigin;
-        transactionContext.ovmGASLIMIT = gasMeterConfig.maxGasPerQueuePerEpoch;
+
+        transactionRecord.ovmGasRefund = 0;
+
+        messageContext.ovmCALLER = address(-1);
+        messageContext.ovmADDRESS = address(-1);
```

Comment:
I think that this is not strictly necessary, but I like the redundancy

Reply:
Discussed offline, we are keeping this, as well as setting them to `address(0)` in `_initContext()`.